### PR TITLE
[DCJ-218][DCJ-208][DCJ-206] Refactor EditConsentGroup.js to JSX

### PR DESCRIPTION
**Pull request by PintoGideon**
### Addresses

https://broadworkbench.atlassian.net/browse/DCJ-218
https://broadworkbench.atlassian.net/browse/DCJ-208
https://broadworkbench.atlassian.net/browse/DCJ-206

### Summary
1. Remove the use of react-hyperscript-helpers
2. Update the file extensions from js to jsx
3. This form was manually tested by going to the `/data_submission_form` route 

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by rushtong — APPROVED** [review]
Looks good! 👍🏽

File: src/pages/data_submission/DataAccessGovernance.jsx [hunk 1]
```diff
@@ -221,61 +220,45 @@ export const DataAccessGovernance = (props) => {
...
-            display: 'flex',
-            alignItems: 'flex-end',
-            margin: '2rem 0 2rem 0'
+  return (
+    <div className='data-submitter-section'>
+      <h2>Data Access Governance</h2>
+      <div>
+        {consentGroupsState?.map((state, idx) => {
+          if (isNil(state)) {
+            return null;
```
Comment (fboulnois) [inline src/pages/data_submission/DataAccessGovernance.jsx]:
```suggestion
            return <div></div>;
```

I think we might want to return an empty div instead to keep it equivalent to the previous code, but this seems to work too.

**Review by fboulnois — APPROVED** [review]
Looks great! One comment:

**Review by aarohinadkarni — APPROVED** [review]
looks good, functions as expected 👍

**Pull request merged**